app/controllers/Checkout.scala

```diff
@@ -1,43 +1,32 @@
 package controllers
 
 import actions.CommonActions._
-import com.gu.identity.play.{IdUser, PrivateFields}
+import com.gu.identity.play.IdUser
 import com.typesafe.scalalogging.LazyLogging
 import forms.{FinishAccountForm, SubscriptionsForm}
-import model.PersonalData
+import model.SubscriptionData
 import play.api.libs.json._
 import play.api.mvc._
 import services.CheckoutService.CheckoutResult
 import services.{CheckoutService, _}
 import views.html.{checkout => view}
 import configuration.Config.Identity.webAppProfileUrl
-
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
 object Checkout extends Controller with LazyLogging {
+  def identityCookieOpt(implicit request: Request[_]): Option[Cookie] =
+    request.cookies.find(_.name == "SC_GU_U")
 
   def renderCheckout = GoogleAuthenticatedStaffAction.async { implicit request =>
-    for (idUserOpt <- getIdentityUserByCookie(request)) yield {
-      def idUserData(keyName: String, fieldName: PrivateFields => Option[String]): Option[(String, String)] =
-        for {
-          idUser <- idUserOpt
-          fields <- idUser.privateFields
-          field <- fieldName(fields)
-        } yield keyName -> field
-
-      val form = SubscriptionsForm().copy(
-        data = (
-          idUserData("personal.first", _.firstName) ++
-          idUserData("personal.last", _.secondName) ++
-          idUserOpt.map("personal.emailValidation.email" -> _.primaryEmailAddress) ++
-          idUserOpt.map("personal.emailValidation.confirm" -> _.primaryEmailAddress) ++
-          idUserData("personal.address.address1", _.billingAddress1) ++
-          idUserData("personal.address.address2", _.billingAddress2) ++
-          idUserData("personal.address.town", _.billingAddress3) ++
-          idUserData("personal.address.postcode", _.billingPostcode)
-        ).toMap
-      )
+    getIdentityUserByCookie.map { idUserOpt =>
+      val form = idUserOpt.map { idUser =>
+        val data = SubscriptionData.fromIdUser(idUser)
+        logger.debug(s"validation failed. Rendering data: $data")
```

Comment:
This is for cases where input passes front-end validation but not back-end validation (hence `BadRequest`). Perhaps we should log as warn instead of debug